openedx/core/djangoapps/content/course_overviews/tests.py

```diff
@@ -485,3 +492,266 @@ def test_get_all_courses_by_org(self):
+        `create_after_overview` is False, we'll create the CourseOverviewImageSet
+        at the same time as the CourseOverview.
+        """
+        # Create a real (oversized) image...
+        image = Image.new("RGB", (800, 400), "blue")
+        image_buff = StringIO()
+        image.save(image_buff, format="PNG")
+        image_buff.seek(0)
+        image_name = "big_course_image.png"
+
+        with self.store.default_store(modulestore_type):
+            course = CourseFactory.create(
+                default_store=modulestore_type, course_image=image_name
+            )
+
+            # Save a real image here...
+            course_image_asset_key = StaticContent.compute_location(course.id, course.course_image)
+            course_image_content = StaticContent(course_image_asset_key, image_name, 'image/png', image_buff)
+            contentstore().save(course_image_content)
+
+            # If create_after_overview is True, disable thumbnail generation so
+            # that the CourseOverview object is created and saved without an
+            # image_set at first (it will be lazily created later).
+            if create_after_overview:
+                self.set_config(enabled=False)
+
+            # Now generate the CourseOverview...
+            course_overview = CourseOverview.get_from_id(course.id)
+
+            # If create_after_overview is True, no image_set exists yet. Verify
+            # that, then switch config back over to True and it should lazily
+            # create the image_set on the next get_from_id() call.
+            if create_after_overview:
+                self.assertFalse(hasattr(course_overview, 'image_set'))
+                self.set_config(enabled=True)
+                course_overview = CourseOverview.get_from_id(course.id)
+
+            self.assertTrue(hasattr(course_overview, 'image_set'))
+            image_urls = course_overview.image_urls
+            config = CourseOverviewImageConfig.current()
+
+            # Make sure the thumbnail names come out as expected...
+            self.assertTrue(image_urls['raw'].endswith('big_course_image.png'))
+            self.assertTrue(image_urls['small'].endswith('big_course_image-png-{}x{}.jpg'.format(*config.small)))
+            self.assertTrue(image_urls['large'].endswith('big_course_image-png-{}x{}.jpg'.format(*config.large)))
+
+            # Now make sure our thumbnails are of the sizes we expect...
+            for image_url, expected_size in [(image_urls['small'], config.small), (image_urls['large'], config.large)]:
+                image_key = StaticContent.get_location_from_path(image_url)
+                image_content = AssetManager.find(image_key)
+                image = Image.open(StringIO(image_content.data))
+                self.assertEqual(image.size, expected_size)
+
+    def _assert_image_url_values(self, modulestore_type, raw_course_image_name, expected_url=None):
```

Comment:
Consider renaming this helper method so it's clear that it's asserting for the failure case.